tools/data_integrity/score_instrument.php

```diff
@@ -0,0 +1,225 @@
+        $params = array_merge(
+            $params,
+            array(
+                'sid' => $sessionID,
+                'cid' => $candID
+            )
+        );
+    }
+    $instrumentMetaData = $DB->pselect($query, $params);
+
+    // return error if no candidates/timepoint matched the args
+    if (!is_array($instrumentMetaData) || empty($instrumentMetaData)) {
+        // given that the tool might be run with all, $candID and $sessionID might not be defined
+        if (!empty($candID) && !empty($sessionID)) {
+            printOut(
+                "No records match the criteria returned for ".
+                "candidate ($candID), timepoint ($sessionID), instrument ($instrumentName)!"
+            );
+        } else {
+            printOut("No records for instrument $instrumentName!");
+        }
+        continue;
+    }
+
+    printOut("Start");
+
+    // loop the list and derive scores for each record
+    foreach ($instrumentMetaData as $record) {
+        // make an instance of the instrument's object
+        $instrument = NDB_BVL_Instrument::factory($instrumentName, $record['CommentID'],'');
+        $instrumentTable = $instrument->table;
+
+        // check if the instrument has a scoring method
+        if (!method_exists($instrument, "score")) {
+            printOut("Error, the instrument ($instrumentName) does not have a scoring feature");
+            continue;
+        }
+
+        // print out candidate/session info
+        printOut("Candidate: ".$record['CandID']."/".$record['Visit_label']."/".$record['SessionID'].":: (".$record['PSCID'].")");
+
+        // call the score function
+        $oldRecord = $DB->pselectRow(
+            "SELECT * FROM $instrumentTable WHERE CommentID=:cid",
+            array('cid' => $record['CommentID'])
+        );
+        $success = $instrument->score();
+        $newRecord = $DB->pselectRow(
+            "SELECT * FROM $instrumentTable WHERE CommentID=:cid",
+            array('cid' => $record['CommentID'])
+        );
+        unset($oldRecord['Testdate']);
+        unset($newRecord['Testdate']);
+        $diff = array_diff_assoc($oldRecord, $newRecord);
+
+        if($diff != array()) {
+
+            logMessage("Updated $record[CommentID]:");
+            foreach ($diff as $key => $val) {
+                $old = $oldRecord[$key] == null ? 'null' : $oldRecord[$key];
+                $new = $newRecord[$key] == null ? 'null' : $newRecord[$key];
+                logMessage("\t$key: $old => $new");
+            }
+        }
+        else {
+            logMessage("No changes made to " . $record['CommentID']);
+        }
+        // unset
+        unset($instrument);
+    }
+}
+fclose($logfp);
+printOut("End");
+
+/*
+ * Prints to log file
+ */
+function logMessage($message) {
+    global $logfp;
+    if (!$logfp) {
+        //The log file could not be instantiated
+        //use print instead
+        print_r($message);
+    }
+    $now_string = strftime("%Y-%m-%d %H:%M:%S");
```

Comment:
need  to specify timezone

Reply:
http://php.net/manual/en/function.date-default-timezone-set.php

Reply:
@ridz1208 Not necessarily, if not, it will use the server time zone. Which is our case is better, you do not want to specify EST for projects in BC or in Europe.

Reply:
@PapillonMcGill It's currently defaulting to UTC so all the dates in the scripts are completely wrong

PHP seems to have deprecated the "default from system" functionality because it was unreliable (see comments on the link above)

Reply:
but I agree... we can't hardcode it in the code thats y I did not make the changes yet... I need to figure it out

Reply:
@ridz1208 in the php.ini set the timezone attribute to local timezone.
date.timezone=America/Toronto as an example

Reply:
that is the other way of doing it but I can't do this for other systems so we don't really have control over it. I wanted to look for a better alternative

Reply:
a timezone config setting is probably the easiest alternative.